### ISSUE TITLE
node-type socket: Add TCP as client and/or server

### ISSUE THREAD

**Issue by al3xa23** [VILLASframework/node#841]
Would be nice to have TCP interface

**Comment by stv0g** [comment]
I agree :) Implementation is a bit more complicated as we need to differentiate between client and server roles.

TCP Performance is also not for real-time..
